File: Leet Code/Hard/330_Patching_Array.cpp
#include<iostream>
#include <vector>
using namespace std;


class Solution {
public:
    int minPatches(vector<int>& nums, int n) {
        long long miss = 1;
        int i = 0;
        int cost = 0;

        while (miss <= n) {
            if (i < nums.size() && nums[i] <= miss) {
                miss += nums[i];
                i++;
            } else {
                miss += miss;
                cost++;
            }
        }

        return cost;
    }
};


int main()
{
    
    return 0;
}

File: Leet Code/Hard/330_Patching_Array.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        miss=1
        cost=0
        i=0

        while(miss<=n):
            if i<len(nums) and nums[i]<=miss:
                miss+=nums[i]
                i+=1
            else:
                miss+=miss
                cost+=1

        return cost
